internal/middleware/ratelimiter.go

```diff
@@ -1,11 +1,83 @@
 package middleware
 
 import (
+	"context"
+	"fmt"
+	"log/slog" // Import the slog package for structured logging
 	"net/http"
+	"strconv"
+	"time"
+
+	dice "github.com/dicedb/go-dice" // Import dice package
 )
 
-func RateLimiter(next http.Handler) http.Handler {
+// RateLimiter middleware to limit requests based on a specified limit and duration
+func RateLimiter(diceClient *dice.Client, next http.Handler, limit, window int) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+		defer cancel()
+
+		// Check DiceDB connection health
+		if err := diceClient.Ping(ctx).Err(); err != nil {
```

Comment:
We don't need to ping DiceDB for dice health, anyways `Get` would send and err if DB is down, we can avoid this extra query to DB.